### PR TITLE
UICONSET-136: add assigned users component

### PR DESCRIPTION
**Pull request by alisher-epam**
## Purpose

[UICONSET-136](https://issues.folio.org/browse/UICONSET-136) - Replicate assigned users functionality on permission sets
## Approach

- Copy all the AssignedUsers component from ui-users and past it inside temp folder
## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [x] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Comment by NikitaSedyx** [comment]
@alisher-epam are you sure you want to add component with issues (ui-users story is not closed)

File: src/temp/AssignedUsers/AssignUsers.css [hunk 1]
```diff
@@ -0,0 +1,9 @@
+.AssignUsersWrapper {
```
Comment (NikitaSedyx) [inline src/temp/AssignedUsers/AssignUsers.css:1]: are all changes in tmp folder 100% the same as in ui-users?

Comment (alisher-epam) [reply src/temp/AssignedUsers/AssignUsers.css:1]: I guess 95% the same. I have updated the constants path and improved code styling according to eslint setup in the module.

Comment (alisher-epam) [reply src/temp/AssignedUsers/AssignUsers.css:1]: @NikitaSedyx ui-users story one-liner fixes included in this PR https://github.com/folio-org/ui-users/pull/2534/files#diff-b4f7d17e02e42601400baccbbc6118e2ea1ce097d40ffc093c028c9f68265ed2R38

Comment (NikitaSedyx) [reply src/temp/AssignedUsers/AssignUsers.css:1]: hm, what have you updated? these files should be 100% the same

Comment (alisher-epam) [reply src/temp/AssignedUsers/AssignUsers.css:1]: @NikitaSedyx you can refer to this PR https://github.com/folio-org/ui-users/pull/2535

Comment (NikitaSedyx) [reply src/temp/AssignedUsers/AssignUsers.css:1]: I can't, you mentioned that it's 95%, what is 5% difference? I can't compare files line by line

Comment (NikitaSedyx) [reply src/temp/AssignedUsers/AssignUsers.css:1]: or do you say that you replicated changes to ui-users as well?

File: src/temp/AssignedUsers/hooks/useAssignedUsers/useAssignedUsers.test.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+describe('useAssignedUsers', () => {
+  const setHeaderMock = jest.fn();
+  const kyMock = jest.fn(() => ({
+    extend: jest.fn(({ hooks: { beforeRequest } }) => {
+      beforeRequest.forEach(handler => handler({ headers: { set: setHeaderMock } }));
+
+      return {
+        get: (path) => ({
+          json: () => Promise.resolve(kyResponseMap[path]),
+        })
```
Comment (usavkov-epam) [inline src/temp/AssignedUsers/hooks/useAssignedUsers/useAssignedUsers.test.js:58]: ESLint - Missing trailing comma

File: src/temp/AssignedUsers/AssignUsers.css (hunk 1 above)
Comment (alisher-epam) [reply src/temp/AssignedUsers/AssignUsers.css:1]: 5% Change was only code styling and all the changes from the PR https://github.com/folio-org/ui-users/pull/2535. Now, it is 100% MATCH.

**Review by NikitaSedyx — APPROVED** [review]

**Review by usavkov-epam — APPROVED** [review]
Update the changelog, please.

File: src/temp/AssignedUsers/hooks/useAssignedUsers/useAssignedUsers.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        acc[group.id] = group.group;
+
+        return acc;
+      }, {});
+
+      return usersResponse.map(({ personal, patronGroup, ...rest }) => ({
+        ...rest,
+        personal,
+        patronGroup,
+        fullName: [personal.firstName, personal.lastName].filter(Boolean).join(' '),
```
Comment (NikitaSedyx) [inline src/temp/AssignedUsers/hooks/useAssignedUsers/useAssignedUsers.js:77]: personal can be null

File: src/temp/AssignedUsers/AssignedUsersList.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  Col,
+  MultiColumnList,
+} from '@folio/stripes/components';
+
+import {
+  COLUMN_MAPPING,
+  COLUMN_WIDTH,
+  VISIBLE_COLUMNS,
+} from './constants';
+import AssignMembers from './AssignUsers';
```
Comment (NikitaSedyx) [inline src/temp/AssignedUsers/AssignedUsersList.js:16]: plz use identical names

File: src/temp/AssignedUsers/AssignedUsersList.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+const AssignedUsersList = ({ users, assignUsers, isFetching }) => (
+  <>
+    <Row end="xs">
```
Comment (NikitaSedyx) [inline src/temp/AssignedUsers/AssignedUsersList.js:20]: it shouldn't be part of `AssignedUsersList`, it should be part of `Accordion`

https://github.com/folio-org/stripes-components/blob/master/lib/Accordion/Accordion.js#L29

File: src/temp/AssignedUsers/AssignUsers.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+  return (
+    <div className={styles.AssignUsersWrapper}>
+      <Pluggable
+        aria-haspopup="true"
+        dataKey="assignUsers"
+        searchButtonStyle="default"
+        searchLabel={buttonLabel}
+        stripes={stripes}
+        type="find-user"
```
Comment (NikitaSedyx) [inline src/temp/AssignedUsers/AssignUsers.js:26]: plugin can't fetch data from selected tenant, fix it

**Pull request merged**